presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/IOPlanPrinter.java

```diff
@@ -88,14 +99,18 @@ public static class IOPlan
     {
         private final Set<TableColumnInfo> inputTableColumnInfos;
         private final Optional<CatalogSchemaTableName> outputTable;
+        private final List<PrestoWarning> warnings;
 
         @JsonCreator
         public IOPlan(
                 @JsonProperty("inputTableColumnInfos") Set<TableColumnInfo> inputTableColumnInfos,
-                @JsonProperty("outputTable") Optional<CatalogSchemaTableName> outputTable)
+                @JsonProperty("outputTable") Optional<CatalogSchemaTableName> outputTable,
+                @JsonProperty("warnings") List<PrestoWarning> warnings)
         {
             this.inputTableColumnInfos = ImmutableSet.copyOf(requireNonNull(inputTableColumnInfos, "inputTableColumnInfos is null"));
             this.outputTable = requireNonNull(outputTable, "outputTable is null");
+            // warnings can be null when deserialized from a json string where they are absent
+            this.warnings = warnings == null ? ImmutableList.of() : ImmutableList.copyOf(warnings);
```

Comment:
I was on the fence about whether to make this field an Optional list instead. I felt it made things a bit clunky (and then you need to add a custom inclusion filter to be able to exclude empty warnings from being serialized).  happy to switch it though, if preferred, as in Presto we do generally prefer to make things optional vs. null.